src/ol/geom/linestring.js

```diff
@@ -203,6 +204,35 @@ ol.geom.LineString.prototype.intersectsExtent = function(extent) {
 
 
 /**
+ * Modifies the geometry's coordinates in place.
+ *
+ * ```js
+ * var lineString = new ol.geom.LineString(coords);
+ *
+ * // to add coordinates at index, do:
+ * lineString.spliceCoordinates(index, 0, [newCoord]);
+ *
+ * // to delete coordinate at index, do:
+ * lineString.spliceCoordinates(index, 1);
+ *
+ * // to replace coordinate at index, do:
+ * lineString.spliceCoordinates(index, 1, [newCoord]);
+ * ```
+ *
+ * @param {number} start Start.
+ * @param {number} deleteCount Delete Count.
+ * @param {Array.<ol.Coordinate>=} opt_coordinates Coordinates.
```

Comment:
Could we improve the API doc a bit? Maybe like this (using the wording from the [MDN docs](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/Array/splice)):

     * @param {number} start Index at which to start changing the coordinates.
     * @param {number} deleteCount An integer indicating the number of old
     *    coordinates to remove. If `deleteCount` is `0`, no elements are removed.
     *    In this case, you should specify at least one new element.
     * @param {Array.<ol.Coordinate>=} opt_coordinates The coordinates to add.
     *    If you don't specify any elements, `spliceCoordinates()` will only remove
     *    elements.